Flaky test: `TestPostToCommunityChat`

# Problem

The test itself passes fine, but panics during `TeardownTest`.
Note the `97.861s` runtime, the test runs quickly, then it hangs for 1-2 mintues and eventually panics.

```nim
goroutine 140931 [running]:
github.com/libp2p/go-libp2p-pubsub.(*Subscription).Topic(...)
        /Users/igorsirotin/Repositories/Status/status-go/vendor/github.com/libp2p/go-libp2p-pubsub/subscription.go:21
github.com/waku-org/go-waku/waku/v2/protocol/relay.(*WakuRelay).unsubscribeFromPubsubTopic(0x14016636200, 0x1401e379880)
        /Users/igorsirotin/Repositories/Status/status-go/vendor/github.com/waku-org/go-waku/waku/v2/protocol/relay/waku_relay.go:506 +0x2c
github.com/waku-org/go-waku/waku/v2/protocol/relay.(*WakuRelay).Unsubscribe(0x14016636200, {0x14001083e48?, 0x102f0af14?}, {{0x1401a73ed80?, 0x1401025be00?}, 0x140166efb30?})
        /Users/igorsirotin/Repositories/Status/status-go/vendor/github.com/waku-org/go-waku/waku/v2/protocol/relay/waku_relay.go:492 +0x2d4
github.com/status-im/status-go/wakuv2.(*Waku).subscribeToPubsubTopicWithWakuRelay.func1()
        /Users/igorsirotin/Repositories/Status/status-go/wakuv2/waku.go:652 +0x230
created by github.com/status-im/status-go/wakuv2.(*Waku).subscribeToPubsubTopicWithWakuRelay in goroutine 135831
        /Users/igorsirotin/Repositories/Status/status-go/wakuv2/waku.go:647 +0x2a4
FAIL    github.com/status-im/status-go/protocol 97.861s
```

# Race condition

Found by go race detector:

```nim
==================
WARNING: DATA RACE
Read at 0x00c000920e90 by goroutine 5273:
  github.com/waku-org/go-waku/waku/v2/protocol/relay.(*broadcaster).UnRegister()
      /Users/igorsirotin/Repositories/Status/status-go/vendor/github.com/waku-org/go-waku/waku/v2/protocol/relay/broadcast.go:218 +0x3c
  github.com/waku-org/go-waku/waku/v2/protocol/relay.(*WakuRelay).unsubscribeFromPubsubTopic()
      /Users/igorsirotin/Repositories/Status/status-go/vendor/github.com/waku-org/go-waku/waku/v2/protocol/relay/waku_relay.go:515 +0x244
  github.com/waku-org/go-waku/waku/v2/protocol/relay.(*WakuRelay).Unsubscribe()
      /Users/igorsirotin/Repositories/Status/status-go/vendor/github.com/waku-org/go-waku/waku/v2/protocol/relay/waku_relay.go:495 +0x730
  github.com/status-im/status-go/wakuv2.(*Waku).subscribeToPubsubTopicWithWakuRelay.func1()
      /Users/igorsirotin/Repositories/Status/status-go/wakuv2/waku.go:657 +0x42c

Previous write at 0x00c000920e90 by goroutine 50:
  github.com/waku-org/go-waku/waku/v2/protocol/relay.(*Subscriptions).close()
      /Users/igorsirotin/Repositories/Status/status-go/vendor/github.com/waku-org/go-waku/waku/v2/protocol/relay/broadcast.go:128 +0x130
  github.com/waku-org/go-waku/waku/v2/protocol/relay.(*broadcaster).Stop()
      /Users/igorsirotin/Repositories/Status/status-go/vendor/github.com/waku-org/go-waku/waku/v2/protocol/relay/broadcast.go:196 +0x64
  github.com/waku-org/go-waku/waku/v2/node.(*WakuNode).Stop()
      /Users/igorsirotin/Repositories/Status/status-go/vendor/github.com/waku-org/go-waku/waku/v2/node/wakunode2.go:506 +0x64
  github.com/status-im/status-go/wakuv2.(*Waku).Stop()
      /Users/igorsirotin/Repositories/Status/status-go/wakuv2/waku.go:1294 +0x124
  github.com/status-im/status-go/protocol.(*MessengerCommunitiesShardingSuite).TearDownTest()
      /Users/igorsirotin/Repositories/Status/status-go/protocol/messenger_communities_sharding_test.go:107 +0x140
  github.com/stretchr/testify/suite.Run.func1.1()
      /Users/igorsirotin/Repositories/Status/status-go/vendor/github.com/stretchr/testify/suite/suite.go:184 +0x26c
  runtime.deferreturn()
      /Users/igorsirotin/go/go1.21.8/src/runtime/panic.go:477 +0x34
  testing.tRunner()
      /Users/igorsirotin/go/go1.21.8/src/testing/testing.go:1595 +0x1b0
  testing.(*T).Run.func1()
      /Users/igorsirotin/go/go1.21.8/src/testing/testing.go:1648 +0x40

Goroutine 5273 (running) created at:
  github.com/status-im/status-go/wakuv2.(*Waku).subscribeToPubsubTopicWithWakuRelay()
      /Users/igorsirotin/Repositories/Status/status-go/wakuv2/waku.go:651 +0x54c
  github.com/status-im/status-go/wakuv2.(*Waku).SubscribeToPubsubTopic()
      /Users/igorsirotin/Repositories/Status/status-go/wakuv2/waku.go:1524 +0xc4
  github.com/status-im/status-go/eth-node/bridge/geth.(*gethWakuV2Wrapper).SubscribeToPubsubTopic()
      /Users/igorsirotin/Repositories/Status/status-go/eth-node/bridge/geth/wakuv2.go:238 +0x50
  github.com/status-im/status-go/protocol/transport.(*Transport).SubscribeToPubsubTopic()
      /Users/igorsirotin/Repositories/Status/status-go/protocol/transport/transport.go:703 +0x80
  github.com/status-im/status-go/protocol.(*Messenger).subscribeToCommunityShard()
      /Users/igorsirotin/Repositories/Status/status-go/protocol/messenger_communities.go:1062 +0xe8
  github.com/status-im/status-go/protocol.(*Messenger).UpdateCommunityFilters()
      /Users/igorsirotin/Repositories/Status/status-go/protocol/messenger_communities.go:2720 +0x534
  github.com/status-im/status-go/protocol.(*Messenger).SetCommunityShard()
      /Users/igorsirotin/Repositories/Status/status-go/protocol/messenger_communities.go:2615 +0x584
  github.com/status-im/status-go/protocol.(*MessengerCommunitiesShardingSuite).testPostToCommunityChat()
      /Users/igorsirotin/Repositories/Status/status-go/protocol/messenger_communities_sharding_test.go:124 +0x344
  github.com/status-im/status-go/protocol.(*MessengerCommunitiesShardingSuite).TestPostToCommunityChat.func1()
      /Users/igorsirotin/Repositories/Status/status-go/protocol/messenger_communities_sharding_test.go:197 +0x60
  github.com/stretchr/testify/suite.(*Suite).Run.func1()
      /Users/igorsirotin/Repositories/Status/status-go/vendor/github.com/stretchr/testify/suite/suite.go:115 +0x1b8
  testing.tRunner()
      /Users/igorsirotin/go/go1.21.8/src/testing/testing.go:1595 +0x1b0
  testing.(*T).Run.func1()
      /Users/igorsirotin/go/go1.21.8/src/testing/testing.go:1648 +0x40

Goroutine 50 (running) created at:
  testing.(*T).Run()
      /Users/igorsirotin/go/go1.21.8/src/testing/testing.go:1648 +0x5e8
  github.com/stretchr/testify/suite.runTests()
      /Users/igorsirotin/Repositories/Status/status-go/vendor/github.com/stretchr/testify/suite/suite.go:247 +0x170
  github.com/stretchr/testify/suite.Run()
      /Users/igorsirotin/Repositories/Status/status-go/vendor/github.com/stretchr/testify/suite/suite.go:220 +0x79c
  github.com/status-im/status-go/protocol.TestMessengerCommunitiesShardingSuite()
      /Users/igorsirotin/Repositories/Status/status-go/protocol/messenger_communities_sharding_test.go:23 +0x40
  testing.tRunner()
      /Users/igorsirotin/go/go1.21.8/src/testing/testing.go:1595 +0x1b0
  testing.(*T).Run.func1()
      /Users/igorsirotin/go/go1.21.8/src/testing/testing.go:1648 +0x40
==================

```

# Notes

Tried to fix it with this, didn't help:
- https://github.com/waku-org/go-waku/pull/1201

Got a new panic now 😵

```nim
<<< WakuRelay.Publish before upsertTopic. topic: /waku/2/rs/16/64
<<< upsertTopic. topic: /waku/2/rs/16/64, found: false, topicaData: <nil>
<<< upsertTopic. added new. topic: /waku/2/rs/16/64, found: false, topicData: &{0x1400b490d20 <nil> <nil> map[]}
<<< WakuRelay.Publish after upsertTopic. pubSubTopic: /waku/2/rs/16/64
2024-08-22T23:17:12.526+0100    WARN    pubsub  go-libp2p-pubsub/tag_tracer.go:122      unable to create decaying delivery tag: decaying tag with name pubsub-deliveries:/waku/2/rs/16/64 already exists
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x10161e9ac]

goroutine 110762 [running]:
github.com/waku-org/go-waku/waku/v2/peermanager.(*PeerManager).checkAndUpdateTopicHealth(0x1400279b8c0, 0x0)
        /Users/igorsirotin/Repositories/Status/status-go/vendor/github.com/waku-org/go-waku/waku/v2/peermanager/peer_manager.go:130 +0x2c
github.com/waku-org/go-waku/waku/v2/peermanager.(*PeerManager).handlerPeerTopicEvent(0x1400279b8c0, {{0x1400e2fe320, 0x10}, {0x14002754420, 0x27}, 0x0})
        /Users/igorsirotin/Repositories/Status/status-go/vendor/github.com/waku-org/go-waku/waku/v2/peermanager/topic_event_handler.go:148 +0x574
github.com/waku-org/go-waku/waku/v2/peermanager.(*PeerManager).peerEventLoop(0x1400279b8c0, {0x104e1bb28, 0x140042b9c20})
        /Users/igorsirotin/Repositories/Status/status-go/vendor/github.com/waku-org/go-waku/waku/v2/peermanager/topic_event_handler.go:173 +0x1e0
created by github.com/waku-org/go-waku/waku/v2/peermanager.(*PeerManager).Start in goroutine 109084
        /Users/igorsirotin/Repositories/Status/status-go/vendor/github.com/waku-org/go-waku/waku/v2/peermanager/peer_manager.go:243 +0xf8
FAIL    github.com/status-im/status-go/protocol 77.783s
FAIL
```

--- a/go.mod
+++ b/go.mod
@@ -95,7 +95,7 @@ require (
 	github.com/schollz/peerdiscovery v1.7.0
 	github.com/siphiuel/lc-proxy-wrapper v0.0.0-20230516150924-246507cee8c7
 	github.com/urfave/cli/v2 v2.27.2
-	github.com/waku-org/go-waku v0.8.1-0.20240819221706-d3b51130599d
+	github.com/waku-org/go-waku v0.8.1-0.20240823143238-949684092ec5
 	github.com/wk8/go-ordered-map/v2 v2.1.7
 	github.com/yeqown/go-qrcode/v2 v2.2.1
 	github.com/yeqown/go-qrcode/writer/standard v1.2.1

--- a/wakuv2/waku.go
+++ b/wakuv2/waku.go
@@ -1850,7 +1850,7 @@ func FormatPeerConnFailures(wakuNode *node.WakuNode) map[string]int {
 	p := make(map[string]int)
 	for _, peerID := range wakuNode.Host().Network().Peers() {
 		peerInfo := wakuNode.Host().Peerstore().PeerInfo(peerID)
-		connFailures := wakuNode.Host().Peerstore().(wps.WakuPeerstore).ConnFailures(peerInfo)
+		connFailures := wakuNode.Host().Peerstore().(wps.WakuPeerstore).ConnFailures(peerInfo.ID)
 		if connFailures > 0 {
 			p[peerID.String()] = connFailures
 		}

